src/clusterfuzz/_internal/system/archive.py

```diff
@@ -407,6 +415,147 @@ class ArchiveType:
   TAR_LZMA = 3
 
 
+@dataclasses.dataclass
+class CacheBlock:
+  """Represents a cache entry for the HttpZipFile.
+  Members:
+    start: the start of the byte range in the file.
+    end: the end of the byte range in the file (inclusive).
+    content: the sequence of bytes for this range.
+  """
+  start: int
+  end: int
+  content: bytes
+
+
+class HttpZipFile(io.IOBase):
```

Comment:
Is all this code original? if it's inspired from some other implementation, maybe it should be linked to?

Reply:
Nope, that's original code I wrote.